File: fsharp-functions/OpenAPI/src/model/UserAllOf.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType

module UserAllOf = 

  //#region UserAllOf

  [<CLIMutable>]
  type UserAllOf = {
    [<JsonProperty(PropertyName = "email")>]
    Email : AnyType;
    [<JsonProperty(PropertyName = "is_bot")>]
    IsBot : AnyType;
    [<JsonProperty(PropertyName = "avatar_url")>]
    AvatarUrl : AnyType;
    [<JsonProperty(PropertyName = "avatar_version")>]
    AvatarVersion : AnyType;
    [<JsonProperty(PropertyName = "full_name")>]
    FullName : AnyType;
    [<JsonProperty(PropertyName = "is_admin")>]
    IsAdmin : AnyType;
    [<JsonProperty(PropertyName = "is_owner")>]
    IsOwner : AnyType;
    [<JsonProperty(PropertyName = "is_billing_admin")>]
    IsBillingAdmin : AnyType;
    [<JsonProperty(PropertyName = "role")>]
    Role : AnyType;
    [<JsonProperty(PropertyName = "bot_type")>]
    BotType : AnyType;
    [<JsonProperty(PropertyName = "user_id")>]
    UserId : AnyType;
    [<JsonProperty(PropertyName = "bot_owner_id")>]
    BotOwnerId : AnyType;
    [<JsonProperty(PropertyName = "is_active")>]
    IsActive : AnyType;
    [<JsonProperty(PropertyName = "is_guest")>]
    IsGuest : AnyType;
    [<JsonProperty(PropertyName = "timezone")>]
    Timezone : AnyType;
    [<JsonProperty(PropertyName = "date_joined")>]
    DateJoined : AnyType;
    [<JsonProperty(PropertyName = "delivery_email")>]
    DeliveryEmail : AnyType;
    [<JsonProperty(PropertyName = "profile_data")>]
    ProfileData : AnyType;
  }
  
  //#endregion
  

File: kotlin-spring/src/main/kotlin/org/openapitools/model/UserAllOf.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.AnyType
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
 * 
 * @param email 
 * @param isBot 
 * @param avatarUrl 
 * @param avatarVersion 
 * @param fullName 
 * @param isAdmin 
 * @param isOwner 
 * @param isBillingAdmin 
 * @param role 
 * @param botType 
 * @param userId 
 * @param botOwnerId 
 * @param isActive 
 * @param isGuest 
 * @param timezone 
 * @param dateJoined 
 * @param deliveryEmail 
 * @param profileData 
 */
data class UserAllOf(

    @field:Valid
    @field:JsonProperty("email") val email: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_bot") val isBot: AnyType? = null,

    @field:Valid
    @field:JsonProperty("avatar_url") val avatarUrl: AnyType? = null,

    @field:Valid
    @field:JsonProperty("avatar_version") val avatarVersion: AnyType? = null,

    @field:Valid
    @field:JsonProperty("full_name") val fullName: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_admin") val isAdmin: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_owner") val isOwner: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_billing_admin") val isBillingAdmin: AnyType? = null,

    @field:Valid
    @field:JsonProperty("role") val role: AnyType? = null,

    @field:Valid
    @field:JsonProperty("bot_type") val botType: AnyType? = null,

    @field:Valid
    @field:JsonProperty("user_id") val userId: AnyType? = null,

    @field:Valid
    @field:JsonProperty("bot_owner_id") val botOwnerId: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_active") val isActive: AnyType? = null,

    @field:Valid
    @field:JsonProperty("is_guest") val isGuest: AnyType? = null,

    @field:Valid
    @field:JsonProperty("timezone") val timezone: AnyType? = null,

    @field:Valid
    @field:JsonProperty("date_joined") val dateJoined: AnyType? = null,

    @field:Valid
    @field:JsonProperty("delivery_email") val deliveryEmail: AnyType? = null,

    @field:Valid
    @field:JsonProperty("profile_data") val profileData: AnyType? = null
) {

}

